components/director/internal/domain/eventing/service.go

```diff
@@ -0,0 +1,188 @@
+package eventing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/label"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/tenant"
+	"github.com/pkg/errors"
+)
+
+const (
+	RuntimeEventingURLLabel = "runtime/event_service_url"
+	EmptyEventingURL        = ""
+)
+
+//go:generate mockery -name=RuntimeRepository -output=automock -outpkg=automock -case=underscore
+type RuntimeRepository interface {
+	List(ctx context.Context, tenant string, filter []*labelfilter.LabelFilter, pageSize int, cursor string) (*model.RuntimePage, error)
+	GetOldestForFilters(ctx context.Context, tenant string, filter []*labelfilter.LabelFilter) (*model.Runtime, error)
+}
+
+//go:generate mockery -name=LabelRepository -output=automock -outpkg=automock -case=underscore
+type LabelRepository interface {
+	GetByKey(ctx context.Context, tenant string, objectType model.LabelableObject, objectID, key string) (*model.Label, error)
+	Upsert(ctx context.Context, label *model.Label) error
+}
+
+type service struct {
+	runtimeRepo RuntimeRepository
+	labelRepo   LabelRepository
+}
+
+func NewService(runtimeRepo RuntimeRepository, labelRepo LabelRepository) *service {
+	return &service{
+		runtimeRepo: runtimeRepo,
+		labelRepo:   labelRepo,
+	}
+}
+
+func (s *service) GetForApplication(ctx context.Context, appID uuid.UUID) (*model.ApplicationEventingConfiguration, error) {
+	_tenant, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	var foundDefault, foundOldest bool
+	runtime, foundDefault, err := s.getDefaultRuntimeForAppEventing(ctx, _tenant, appID)
+	if err != nil {
```

Comment:
if 2 Runtimes have label `app1/defaultEventing`, we will return randomly one of them, is it ok?

Reply:
We cannot block this case as we do not support the uniqueness of the labels. I will return the error when there are multiple runtimes with the label.